lib/Fmt_ast.ml

```diff
@@ -1714,19 +1724,20 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
           ~break
       in
       let fmt_grp grp = list grp " ;@ " (fmt_expression c) in
-      hvbox 0
-        (Params.parens_if parens c.conf
-           ( hvbox c.conf.fmt_opts.extension_indent.v
-               (wrap "[" "]"
-                  ( str "%"
-                  $ hovbox 2
-                      ( fmt_str_loc c name $ str " fun "
-                      $ fmt_attributes c ~suf:" " call.pexp_attributes
-                      $ fmt_fun_args c xargs $ fmt_opt fmt_cstr $ fmt "@ ->"
-                      )
-                  $ fmt "@ " $ fmt_expression c xbody ) )
-           $ fmt "@ ;@ "
-           $ list grps " ;@;<1000 0>" fmt_grp ) )
+      pro
```

Comment:
Ideally the `pro` would be pushed as far as possible into the boxes (and perhaps the opening parenthese should be appended to the `pro`) and the outer box should be removed.
But this is out of the scope of this PR.